### PR TITLE
RPCh Alpha integration

### PR DESCRIPTION
**Pull request by diegoalzate**
Fixes https://github.com/Rpc-h/extension-block-wallet/issues/7

### CODE REVIEW

**Comment by diegoalzate** [comment]
For `RPCh/SDK` to work with `web` it requires `document` variable. Which is a problem with because now with manifest 3 it looks like it no longer has access to the dom. Plus, the node version of `sdk` is ideal because it would require polyfills.

**Comment by diegoalzate** [comment]
UI issues:
- When doing a swap it is possible that Block wallet throws a "gas estimation failed", after clicking "ok" the transaction can still go through
image
- Sometimes the activity tab in bw may not show the transaction, if this is the case you can switch networks from "Gnosis on RPCh" to another network and then back again to force the ui to update.
CleanShot 2023-02-22 at 15 36 58
- If you encounter a "Network Provider Down" error page you may also have to switch network and back again to get "Gnosis on RPCh" online again. (This can happen when restarting chrome or restarting the wallet)
CleanShot 2023-02-22 at 15 40 23
- Blockwallet can sometimes get a error screen that looks something like the image below. If that is the case you can restart the wallet and possibly have to to get switch network and back again to get "Gnosis on RPCh" online again.
CleanShot 2023-02-22 at 15 39 28

**Comment by ItsLuized** [comment]
- When a transaction is performed ( a swap ) and it's already confirmed the Activity tab shows the transaction still processing. ( updates after some seconds )

File: packages/background/src/utils/constants/networks.ts [hunk 1]
```diff
@@ -411,6 +411,37 @@ export const INITIAL_NETWORKS: Networks = {
...
+        ens: false,
+        showGasLevels: false,
+        iconUrls: [
+            'https://raw.githubusercontent.com/block-wallet/assets/master/blockchains/xdai/assets/0x/logo.png',
+        ],
+        rpcUrls: ['https://primary.gnosis-chain.rpc.hoprtech.net'],
+        blockExplorerUrls: ['https://blockscout.com/xdai/mainnet'],
+        blockExplorerName: 'Blockscout',
+        etherscanApiUrl: 'https://api-gnosis.etherscan.io',
+        actionsTimeIntervals: { ...ACTIONS_TIME_INTERVALS_DEFAULT_VALUES },
```
Comment (julianariel) [inline packages/background/src/utils/constants/networks.ts:438]: @diegoalzate changing this line will make transactions get confirmed quicker as it'll reduce the intervals that the extension waits to perform some actions like block pulling
```suggestion
        actionsTimeIntervals: { ...FAST_TIME_INTERVALS_DEFAULT_VALUES },
```

File: packages/background/rpch/apply_patch.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+const fs = require('fs');
+const path = require('path');
+const DIST_DIR = path.join(__dirname, '..', '..', '..', 'dist');
+
+function patchBackgroundJs() {
+    console.log('[RPCh] Applying patch for background.js');
+    const filePath = path.join(DIST_DIR, 'background.js');
+    const content = fs.readFileSync(filePath, 'utf-8');
+    const updatedContent = content.replace(
```
Comment (julianariel) [inline packages/background/rpch/apply_patch.js:13]: this patch doesn't work when the build is created in production mode (minified and without source maps). You can try this by running the `make build/prod` command

Comment (ItsLuized) [reply packages/background/rpch/apply_patch.js:13]: We will change to patching the minified version instead. Thanks for pointing it out

**Pull request closed without merging**